### PR TITLE
Factor out guts of `dstrace` into new `info` function

### PR DESCRIPTION
**Pull request by bpbond**
This lets users easily get information about just a single object. See examples in `?info`

### CODE REVIEW

**Comment by bpbond** [comment]
@swd-turner if you have time. Thanks. If not, no worries, just reassign to someone else :)

**Comment by swd-turner** [comment]
@bpbond sure, I'll take a look later this week.

File: R/dstrace.R [hunk 1]
```diff
@@ -153,8 +101,8 @@ dstrace <- function(object_name, direction = "upstream", graph = FALSE,
 #'
 #' @param object_name Name of original object being traced
 #' @param tracelist Record of the trace, a tibble
-#' @param upstream Upstream? Logical
-#' @param downstream Downstream? Logical
+#' @param upstream Looking upstream? Logical
+#' @param downstream Looking downstream? Logical
```
Comment (swd-turner) [inline R/dstrace.R]: Took me a bit to figure out what "looking upstream" / "looking downstream" meant. Maybe "Return upstream precursors?" and "Return downstream dependencies?"

File: R/dstrace.R [hunk 2]
```diff
@@ -196,3 +144,116 @@ dstrace_plot <- function(object_name, tracelist, upstream, downstream, ...) {
...
+#' Print and return information about an object.
+#'
+#' @param object_name Name of object to get information about (can be either a data object or a code chunk)
+#' @param gcam_data_map A tibble of metadata information; normally a built-in package dataset
+#' @param previous_tracelist Information about previous objects printed (if called from \code{\link{dstrace}})
+#' @param upstream Looking upstream? Logical
+#' @param downstream Looking downstream? Logical
+#' @return If called directly, returns an entry from \code{GCAM_DATA_MAP}; if called from \code{\link{dstrace}}, a two-
+#' element list with linked object and tracelist information. If called and \code{object_name} if the name of a
+#' code chunk (as opposed to a data object), the relevant help page will be pulled up.
```
Comment (swd-turner) [inline R/dstrace.R]: typo here? If called and object_name _is_ the name of a code chunk

**Comment by bpbond** [comment]
@swd-turner Cool, changes made. Let me know if you have any more, and/or approve when ready. Thanks!

**Comment by swd-turner** [comment]
I was thinking that an object search function would be a useful addition here. Something that looks through `GCAM_DATA_MAP` for all chunks and/or data objects that contain "water" or "ag" etc in the title... e.g., `search_objects("water", type = chunk)`

**Comment by bpbond** [comment]
@swd-turner Huh, interesting. I'm going to move this to its own issue, as it may need some thinking/fleshing out.

**Pull request merged**